### PR TITLE
Rewrite the query to fetch an organization.

### LINKED ISSUE #964: Bulk import workflow encounters IntegrityError when saving an organization
Currently, we allow organizations with the domain that contains the `www` subdomain. For eg: Google can exist as `www.google.com` as well as `google.com`, leading to `IntegrityError` while saving the first while the second exists.

Shouldn't we enforce one URL pattern and trim/add `www` to the `domain` field when saving an organization?

Testcase:

``` py
In [5]: Organization.objects.create(fullname='Google', domain='google.com')
Out[5]: <Organization: google.com>

In [6]: Organization.objects.create(fullname='Google', domain='www.google.com')
---------------------------------------------------------------------------
IntegrityError                            Traceback (most recent call last)
```

### PR DESCRIPTION
**Pull request by narayanaditya95**
Prevents IntegrityError while trying to save an organization
with conflicting fullname but different domains.
Fixes #964

### CODE REVIEW

File: pydata/api.py [hunk 1]
```diff
@@ -118,14 +119,19 @@ class SponsorshipAPIClient(BaseAPIClient):
     model = Sponsorship
 
     def parse(self, sponsor):
+        domain = urlparse(sponsor['external_url']).netloc
+        try:
+            organization = Organization.objects.filter(
+                Q(fullname=sponsor['name']) | Q(domain=domain)
+            ).first()
```
Comment (chrismedrela) [inline pydata/api.py]: `.first()` returns the first object matched by the queryset, **or None if there is no matching object**.

**Comment by chrismedrela** [comment]
@narayanaditya95 , does this PR really fixes #964, that is, can we close #964 when we merge this PR?

**Comment by narayanaditya95** [comment]
@chrismedrela Yes, this PR fixes #964.

I will be creating a new PR for tests as it involves a change to `travis.yml`.

**Comment by chrismedrela** [comment]
> I will be creating a new PR for tests as it involves a change to travis.yml.

Why is a change to `travis.yml` necessary?

**Comment by narayanaditya95** [comment]
@chrismedrela 
The test suite tests the project without adding `pydata` to the list of installed apps. I will create a new job that makes changes to the codebase necessary for including the `pydata` app within the project and then run the test suite. Also, it would involve marking certain tests as expected failures when run with `pydata` installed.

So, the current plan of action is to create three jobs on travis:
1. Runs the test suite without making any changes.
2. Merges the branch `pydata-deploy` and then runs the test suite.
3. Runs the check for missing migrations.

**Pull request merged**